src/twisted/test/test_main.py

```diff
@@ -19,12 +19,21 @@
 class MainTests(TestCase):
     """Test that twisted scripts can be invoked as modules."""
 
+    # Force the full diff to make debugging easier on failures.
+    maxDiff = None
+
     def test_twisted(self):
         """Invoking python -m twisted should execute twist."""
         cmd = sys.executable
         p = Accumulator()
         d = p.endedDeferred = defer.Deferred()
-        reactor.spawnProcess(p, cmd, [cmd, "-m", "twisted", "--help"], env=None)
+        # Enforce 80 columns as otherwise the test are failing when executed in debug mode
+        # or other places in which COLUMNS env var is defined.
+        # The environment is set for both parent and child processes.
+        self.patch(os, "environ", dict(os.environ, COLUMNS="80"))
+        reactor.spawnProcess(
+            p, cmd, [cmd, "-m", "twisted", "--help"], env=dict(os.environ, COLUMNS="80")
+        )
```

Comment:
It would probably be nice to factor out the duplicate `dict(...)` since the point is to make the environment identical in the parent and child (using the same object makes this clearer and more likely).

Reply:
Based on the conversation over IRC I went and simplified the assertion to not check for the full help message.

I hope this is better :)